compiler/rustc_const_eval/src/interpret/intern.rs

```diff
@@ -318,6 +329,40 @@ pub enum InternKind {
     Promoted,
 }
 
+pub(crate) fn take_static_root_alloc<
+    'mir,
+    'tcx: 'mir,
+    M: CompileTimeMachine<'mir, 'tcx, const_eval::MemoryKind>,
+>(
+    ecx: &mut InterpCx<'mir, 'tcx, M>,
+    alloc_id: AllocId,
+) -> Allocation {
+    ecx.memory.alloc_map.remove(&alloc_id).unwrap().1
+}
+
+pub(crate) fn create_static_alloc<
```

Comment:
I don't see why these functions are in `intern.rs`, they don't work on the global tcx memory at all I think?